### DISCUSSION TITLE
채널리스트 이슈와 몇가지 궁금한 점이 있습니다.

### DISCUSSION THREAD

**Discussion by h2saj5xz** [storycraft/node-kakao#223] [Q&A]
안녕하세요.

며칠 개발하면서 겪은 채널리스트 이슈와 
몇가지 궁금한점들이 있어서 글을 남깁니다.

# 채널리스트

대화방이 100개를 초과하는 계정에서
channelList.size를 쳐보면 결과값이 100이 출력됩니다.
이 경우 전체 대화방중 일부에서만 chat 이벤트가 발생합니다.

클라이언트가 처음 불러온 100개만 이벤트가 발생하는 것 같은데
나머지를 불러올 수 있는 방법은 없는건가요?

4.1.1 에서 추가된 channel_added 이벤트가 발생한적도 있지만
새로 추가한 친구가 보낸 메시지에서는 작동하지 않았습니다.

# 센드미디어

4.0.x 버전일때 작동했던 사진 보내는 코드가 작동하지 않아서
example 코드처럼 구성하고 보내봤는데도 작동하지 않습니다.

버전을 4.0.x로 다운하고 테스트해봐도 작동하지 않는데,
제가 시도해볼만한 것들이 있는지,
다른분들은 정상 작동하는지 궁금합니다.

# chat 이벤트

센드미디어가 작동하지 않아서 지금은 확인할 수 없지만
전에 센드미디어시 chat 이벤트가 발생했었는데,
센드챗은 chat 이벤트가 발생하지 않네요.

노드카카오에서 센드챗이나 센드미디어로 챗을 보내면
chat 이벤트가 발생하지 않는게 정상 동작인지 궁금합니다.

**Comment by ghost** [comment]
센드미디어

sendMedia는 정상적으로 작동합니다.
사용한 코드는 다음과 같습니다. (attachment를 upload해서 보내는건 테스트용으로 넣어놨습니다.)

```ts
client.on("chat", async (data: TalkChatData, channel: TalkChannel) => {
	if (data.text === "!test") {
		const attachment = await AttachmentApi.upload(
			2,
			"image",
			readFileSync("./image.png")
		);
		const builder = new ChatBuilder();
		if (attachment.success) {
			builder.attachment(attachment.result);
			void channel
				.sendChat(builder.build(2))
				.then((e) => console.log(e));
			void channel
				.sendMedia(2, {
					data: readFileSync("./image.png"),
					width: 100,
					height: 100,
					name: "image"
				})
				.then((e) => console.log(e));
		}
	}
});
```

**Reply by ghost** [reply]
chat 이벤트

저도 chat 이벤트는 안오네요.

**Reply by h2saj5xz** [reply]
확인 감사드립니다. 참고해서 다시한번 테스트해 보겠습니다.

**Comment by storycraft** [comment]
## 채널 리스트
로그인 직후 초기 채팅방은 서버에서 100개만 제공 되는것으로 보입니다.
일부 패킷 핸들링이 제대로 이루어지지 않는 버그를 발견해 4.1.2에서 픽스했습니다. 업데이트를 할 경우 정상작동합니다.

## sendMedia
정상적으로 작동합니다. 윗분의 예제처럼 AttachmentApi를 사용해도 됩니다.

## chat 이벤트
발생하지 않는 것이 정상입니다

**Reply by h2saj5xz** [reply]
와 4.1.2 나왔네요. 이미 확인하고 개발중이셨군요. 
chat 이벤트도 미디어랑 챗이랑 둘다 발생하지않는지 다시 확인해보겠습니다.
감사합니다.

**Reply by cjh980402** [reply]
채널 리스트 관련해서 질문이 있습니다. 초기 채팅방은 서버에서 100개만 제공 되는데, 메시지가 오기 전에 채널 ID로  채널을 추가할 수 있는 방법이 있나요?

**Reply by storycraft** [reply]
TalkChannelList 객체의 open, normal 속성에서 addChannel 메서드를 사용해  특정 id 채널을 각각 오픈 채팅방과 일반 채팅방으로 추가 할 수 있습니다.

**Reply by cjh980402** [reply]
답변 감사합니다.
